File: 2419-longest-subarray-with-maximum-bitwise-and/2419-longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int count = 1;
        int maxVal = nums[0];
        int ans=1;
        int val=nums[0];
        for(int i=1;i<=nums.size();i++){
            if(i<nums.size() && nums[i]==nums[i-1]){
                count++;
            } else {
                if(val>maxVal){
                    ans=count;
                    maxVal=val;
                } else if(val==maxVal){
                    ans=max(ans,count);
                }
                if(i==nums.size()){
                    break;
                }
                val=nums[i];
                count=1;
            }
        }
        return ans;
    }
};



/*

1    2   3   3   2   2   

    0   0   0   1
    0   0   1   0
    0   0   1   1
    0   0   1   1
    0   0   1   0
    0   0   1   0
-----------------------
    0   0   0   0



*/

File: 2419-longest-subarray-with-maximum-bitwise-and/2419-longest-subarray-with-maximum-bitwise-and.java
class Solution {
    public int longestSubarray(int[] nums) {
        int k = Arrays.stream(nums).max().getAsInt();
        int ans = 1;
        int count = 0;
        for(int i=0;i<=nums.length;i++) {
            if(i<nums.length && nums[i]==k) count++;
            else {
                ans = Math.max(ans,count);
                count=0;
            }
        }
        return ans;
    }
}